### PR TITLE
Replace custom configure options with default build types

### PR DESCRIPTION
**Pull request by kkiesling**
This PR uses common build types from cmake instead of custom options. The "debug" and "optimize" options were removed as they are redundant with setting `-DCMAKE_BUILD_TYPE=Debug` and `-DCMAKE_BUILD_TYPE=Release`, respectively. Previously, the `debug` option also set `-O0` where as specifying the `Debug` build type doesn't do that (instead there is no `-O#` set). With the `Release` type, the additional flag `-DNDEBUG` is also now set.

Also in this PR is the removal of the variable `cxxflags`. It looks like that was just used as a variable to maintain a list that then set the compile flags at the end. This is also redundant with using cmake's `add_compile_options`.

To view what flags are being used at compile time, you can use `-DCMAKE_EXPORT_COMPILE_COMMANDS=ON` and see the file `compile_commands.json`. I did confirm that the flags were still being set with the above changes.

### CODE REVIEW

**Comment by kkiesling** [comment]
One additional thought: If people are used to using the `-debug` and `-optimize` flags, I can add those back in as options and instead just set the `CMAKE_BUILD_TYPE` appropriately rather than setting the flags explicitly.

File: CMakeLists.txt
Comment (paulromano) [inline CMakeLists.txt:114]: The "modern" CMake paradigm is to do everything based on [targets](https://gist.github.com/mbinna/c61dbb39bca0e4fb7d1f73b0d66a4fd1#targets-and-properties). See the recommendation [here](https://gist.github.com/mbinna/c61dbb39bca0e4fb7d1f73b0d66a4fd1#forget-the-commands-add_compile_options-include_directories-link_directories-link_libraries) to not use add_compile_options.

File: CMakeLists.txt
Comment (paulromano) [inline CMakeLists.txt:136]: Perhaps rather than setting this default -O2, we ought to just set `CMAKE_BUILD_TYPE` to `Release` if it's not explicitly set. Note that you'll have to set it as a [cache variable](https://cmake.org/cmake/help/latest/command/set.html#set-cache-entry) to get it to persist.

**Review by paulromano — CHANGES_REQUESTED** [review]
Thanks for taking a first shot at this @kkiesling! Other than the line comments, we'll need a few updates in other places:

- Documentation (docs/source/usersguide/install.rst, troubleshoot.rst)
- Dockerfile
- CI scripts (tools/ci/gha-install.py)

File: CMakeLists.txt
Comment (kkiesling) [reply CMakeLists.txt:114]: Ah interesting- I will remove this change then

File: CMakeLists.txt
Comment (kkiesling) [reply CMakeLists.txt]: I might not have been setting this right, but I got it to work with an if-statement and no cache.

**Comment by kkiesling** [comment]
For the documentation- do you want me to add information about setting the build type or is it assumed that people know `Release` and `Debug` are always options with cmake?

**Review by paulromano — CHANGES_REQUESTED** [review]
For the documentation piece, it would be good to explicitly say that OpenMC respects `CMAKE_BUILD_TYPE` (especially for people who were already familiar with -Ddebug and -Dopenmp).

For the other flags, we should probably rename to be in line with common practice:
- `profile` → `OPENMC_ENABLE_PROFILE`
- `coverage` → `OPENMC_ENABLE_COVERAGE`
- `openmp` → `OPENMC_USE_OPENMP`
- `dagmc` → `OPENMC_USE_DAGMC`
- `libmesh` → `OPENMC_USE_LIBMESH`

**Comment by kkiesling** [comment]
I am a bit lost on how the tests/CI failed on that one build as the result of these changes..

**Comment by paulromano** [comment]
@kkiesling It looks like the tests that rely on finding the installed OpenMC are failing when using DAGMC/libmesh. I think you'll need to update the following:
https://github.com/openmc-dev/openmc/blob/73ba908f1fd8ba239412703142fdbf8bcf06efd8/cmake/OpenMCConfig.cmake.in#L8-L17
Namely, `@dagmc@` should now be `@OPENMC_USE_DAGMC@` and `@libmesh@` should be `@OPENMC_USE_LIBMESH@`.

**Comment by kkiesling** [comment]
Ah thank you for finding that location!

File: CMakeLists.txt
Comment (paulromano) [inline CMakeLists.txt:30]: Looks like something got messed up on these lines (descriptions don't match the variables)

**Review by paulromano — CHANGES_REQUESTED** [review]

File: CMakeLists.txt
Comment (kkiesling) [reply CMakeLists.txt]: lol ya I know how that happened my bad 🤦‍♀️

File: CMakeLists.txt
Comment (paulromano) [inline CMakeLists.txt:35]:
```suggestion
# Set a default build configuration if not explicitly specified
```

File: CMakeLists.txt
Comment (paulromano) [inline CMakeLists.txt:39]:
```suggestion
if(NOT CMAKE_BUILD_TYPE)
    message(STATUS "No build type selected, defaulting to RelWithDebInfo")
    set(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING "Choose the type of build" FORCE)
```

**Review by paulromano — CHANGES_REQUESTED** [review]
A few more small nitpicks and then I think this is good to go. Thanks!

**Review by paulromano — APPROVED** [review]
Great! will merge once I get the green light

**Pull request merged**
